src/login/logind-session.c

```diff
@@ -716,7 +716,7 @@ static int session_dispatch_stop_on_idle(sd_event_source *source, uint64_t t, vo
                 return session_stop(s, /* force */ true);
         }
 
-        r = sd_event_source_set_time(source, usec_add(ts.monotonic, s->manager->stop_idle_session_usec));
+        r = sd_event_source_set_time(source, usec_add(ts.monotonic ?: now(CLOCK_MONOTONIC), s->manager->stop_idle_session_usec));
```

Comment:
`dual_timestamp_is_set()` is better? Though, this change should not be necessary if the above comment is addressed.

Reply:
Good point!